### PR TITLE
Allow user signin using email

### PR DESCRIPTION
**Pull request by FadhlanR**
Ticket: CS-6343

https://github.com/cardstack/boxel/assets/12637010/f6d01e9c-1b91-444e-acdf-630a4b09c568

### CODE REVIEW

File: packages/host/app/services/matrix-service.ts [hunk 1]
```diff
@@ -407,6 +407,55 @@ export default class MatrixService extends Service {
...
+      let error = new Error(data.error) as any;
+      error.data = data;
+      error.status = response.status;
+      throw error;
+    }
+  }
+
+  async login(usernameOrEmail: string, password: string) {
+    try {
+      const cred = await this.client.loginWithPassword(usernameOrEmail, password);
```
Comment (jurgenwerk) [inline packages/host/app/services/matrix-service.ts]: Is this supposed to be `loginWithUsername`?

Comment (FadhlanR) [reply packages/host/app/services/matrix-service.ts]: that function is from matrix client

Comment (jurgenwerk) [reply packages/host/app/services/matrix-service.ts]: I see... it looked confusing to me to have `loginWithPassword` and `loginWithEmail` while password is actually needed in both cases

Comment (habdelra) [reply packages/host/app/services/matrix-service.ts:447]: so the idea is that `loginWithPassword()` uses a password for authorization. loginWithEmail() uses email for authorization: a magic link sent to your email that when clicked on let's you login without a password

Comment (jurgenwerk) [reply packages/host/app/services/matrix-service.ts]: 👍

**Review by habdelra — APPROVED** [review]
great job!

**Pull request merged**